subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -75,7 +75,7 @@ static azure_iot_hub_evt_handler_t evt_handler;
  * name + device ID length + lengths of ".azure-devices-provisioning.net/"
  * and "/?api-version=2018-06-30"
  */
-#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS)
+#if IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS) || IS_ENABLED(CONFIG_AZURE_IOT_HUB_DEVICE_ID_APP)
 #define USER_NAME_BUF_LEN	(CONFIG_AZURE_IOT_HUB_HOSTNAME_MAX_LEN + \
```

Comment:
I think it is OK that the buffer size includes `.azure-devices-provisioning.net/`, more correct would be to have two different `USER_NAME_BUF_LEN` definitions.

Reply:
Thanks for the comments! I just pushed an updated version with the define split up and updated comments